akka-http/src/main/scala/akka/http/javadsl/server/directives/BasicDirectives.scala

```diff
@@ -128,6 +131,13 @@ abstract class BasicDirectives {
     D.extractExecutionContext { c ⇒ inner.apply(c).toScala })
 
   /**
+   * Extracts a single value using the given function.
+   */
+  def extract[T](extract: JFunction[RequestContext, T], inner: JFunction[T, Route]): Route = ScalaRoute {
```

Comment:
are you actually planning to extract a whole Java requestcontext? I didn't do that, since it would've added a lot of Java work (that part is really scala-only). I couldn't see any real use cases for it, even though it would of course be needed if you want 100% feature parity. It would also require introducing RouteResult on the Java side.